client/utils/utils.go

```diff
@@ -99,6 +100,49 @@ func PrintUnsignedStdTx(txCtx authctx.TxContext, cliCtx context.CLIContext, msgs
 	return
 }
 
+// SignStdTx appends a signature to a StdTx and returns a copy of a it. If appendSig
+// is false, it replaces the signatures already attached with the new signature.
+func SignStdTx(txCtx authctx.TxContext, cliCtx context.CLIContext, name string, stdTx auth.StdTx, appendSig bool) (auth.StdTx, error) {
+	var signedStdTx auth.StdTx
+
+	keybase, err := keys.GetKeyBase()
+	if err != nil {
+		return signedStdTx, err
+	}
+	info, err := keybase.Get(name)
+	if err != nil {
+		return signedStdTx, err
+	}
+	addr := info.GetPubKey().Address()
+
+	// Check whether the address is a signer
+	if !isTxSigner(sdk.AccAddress(addr), stdTx.GetSigners()) {
+		fmt.Fprintf(os.Stderr, "WARNING: The generated transaction's intended signer does not match the given signer: '%v'", name)
```

Comment:
Can you think of a case where the user would want to do this? Not sure why we would even allow them to proceed here.

Reply:
In line of principle I agree, though te thought that it was enough to print out a warning

Reply:
Furthermore, I think that offline generation/signing should not bother too much about tx validation as their outputs are JSON files that could be edited at a later stage anyway. It'd be nice to hear thoughts from @jackzampolin too

Reply:
Agree. I don't think validation on generation is super important.

Reply:
Agree. I don't think validation on generation is super important.